oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/FileStoreBuilder.java

```diff
@@ -396,6 +401,17 @@ public FileStoreBuilder withEagerSegmentCaching(boolean eagerSegmentCaching) {
         this.eagerSegmentCaching = eagerSegmentCaching;
         return this;
     }
+    
+    /**
+     * Sets the threshold under which binaries are inlined in data segments.
+     * @param binariesInlineThreshold the threshold
+     * @return this instance
+     */
+    public FileStoreBuilder withBinariesInlineThreshold(int binariesInlineThreshold) {
+        checkArgument(binariesInlineThreshold <= Segment.MEDIUM_LIMIT);
```

Comment:
I would add:
checkArgument(binariesInlineThreshold >= 0);

Reply:
I added that check as well, thanks for chiming in!